src/main/kotlin/org/sol4k/VersionedTransaction.kt

```diff
@@ -0,0 +1,85 @@
+package org.sol4k
+
+import okio.Buffer
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.Base64
+import kotlin.math.max
+
+class VersionedTransaction(
+    val message: Message,
+    val signatures: MutableList<String>,
+) {
+
+    fun sign(keypair: Keypair) {
+        val data = message.serialize()
+        val signature = keypair.sign(data)
+        for (i in 0 until message.header.numRequireSignatures) {
+            val a = message.accounts[i]
+            if (a.verify(signature, data)) {
+                if (signatures.isEmpty()) {
+                    signatures.add(Base58.encode(signature))
+                } else {
+                    signatures[i] = Base58.encode(signature)
+                }
+                break
+            }
+        }
+    }
+
+    fun serialize(): ByteArray {
+        if (signatures.isEmpty() || signatures.size != message.header.numRequireSignatures) {
+            throw Exception("Signature verification failed")
+        }
+
+        val messageData = message.serialize()
+
+        val b = Buffer()
+        b.write(Binary.encodeLength(signatures.size))
+        for (s in signatures) {
+            b.write(Base58.decode(s))
+        }
+        b.write(messageData)
+        return b.readByteArray()
+    }
+
+    fun calcFee(): BigDecimal {
+        val sigFee = lamportToSol(BigDecimal(5000 * max(signatures.size, 1)))
```

Comment:
Why is it 5000? I feel like the `calcFee` function doesn't need to be a part of the package (might be mistaken though) and should be a pert of the client code using VersionedTransaction. What do you think?

Reply:
According some docs [calculating-transaction-fees](https://solana.com/docs/core/fees#calculating-transaction-fees) and [fees](https://solana.com/news/evm-to-svm#fees)

I think this is a useful feature, as for whether to be a part of the VersionedTransaction, we can think about it. I know that rust sdk has a [FeeCalculator](https://docs.rs/solana-sdk/latest/solana_sdk/fee_calculator/struct.FeeCalculator.html)

Reply:
Ok, then let's keep it. I would suggest two changes to it:

- Let's not hardcode 5000 and allow the clients to pass `lamportsPerSignature` as an input argument of the function
- Let's rename the function to `calculateFee`